src/index.ts

```diff
@@ -44,19 +46,28 @@ const loadMovies = async (searchTerm: string): Promise<void> => {
   }
 }
 
-async function searchMovies(searchTerm: string): Promise<void> {
-  if (searchTerm.length > 0) {
-    await loadMovies(searchTerm)
-  }
-}
-async function keyPressHandler(e: KeyboardEvent): Promise<void> {
+async function searchMovies(e: KeyboardEvent): Promise<void> {
   if (e.key === 'Enter') {
     const searchTerm = movieSearchBox.value.trim()
+    movieSearchBox.value = ''
     movieDisplay.innerHTML = ''
-    await searchMovies(searchTerm)
+
+    if (!searchTerm.length) {
+      e.preventDefault()
```

Comment:
not sure if this is necessary here, I don't think the default key event does anything we want to stop. `preventDefault()` tells the _browser_ not to do something, it doesn't tell _our code_ not to do something. Usually this is used with a `form` `submit` event because the _browser_ has default behavior when forms are submitted that developers often want to avoid.

Reply:
Thanks for breaking this down, I thought was specific to the 'event' that that was passed as parameter. I'll remove it.

Reply:
it is specific to the event that was passed in. In this case you're telling the browser to stop the default behavior of the key press event. But there is no real behavior for a key press event, so there's nothing to stop. I just gave the `form submit` event as an example, because unlike the `keypress event`, a form submission _does_ have meaningful behavior (the browser will try to load the URL in the `<form>` `action` attribute. See https://developer.mozilla.org/en-US/docs/Learn/Forms/Sending_and_retrieving_form_data and https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event.